libminifi/src/io/tls/TLSSocket.cpp

```diff
@@ -182,9 +187,19 @@ int16_t TLSSocket::initialize(bool blocking) {
     setNonBlocking();
   logger_->log_trace("Initializing TLSSocket %d", is_server);
   int16_t ret = context_->initialize(is_server);
-  Socket::initialize();
 
-  if (!ret && listeners_ == 0) {
+  if (ret != 0) {
+    logger_->log_warn("Failed to initialize SSL context!");
+    return -1;
+  }
+
+  ret = Socket::initialize();
+  if (ret != 0) {
+    logger_->log_warn("Failed to initialise basic socket for TLS socket");
+    return -1;
```

Comment:
`context_` is left initialized even if socket initialization has failed. This is not changed behavior, but is it intended?

Speaking about suspicious behavior, `TLSSocket::logger_` will be `nullptr` if the object is initialized via the move ctor.

Reply:
I was wrong about the `logger_` part, since both `Socket` and `TLSSocket` have their own separate private `logger_`s, so move will work.

Reply:
Unified loggers to only have one.
Made context init reentrant.